### ISSUE TITLE
What do you use to run code?

### ISSUE THREAD

**Issue by somnathrakshit** [Gotham13121997/superCodingBot#3]
Do you run the code in your private server or do you use some other website through an API that runs the code and sends you the result?

**Comment by cB-Abhinav-Gautam** [comment]
I have hosted this on openshift online. You can run it anywhere with a python environment . The python-telegram-bot library takes care of getting the data from telegram's server to you.

**Comment by cB-Abhinav-Gautam** [comment]
for the /compiler command I use hackerranks api. And for /upcoming , /ongoing I use clists api

**Comment by somnathrakshit** [comment]
My main question is regarding the /compile command since Hackerrank [says](https://www.hackerrank.com/api/docs) that they've discontinued the API.

**Comment by cB-Abhinav-Gautam** [comment]
Yeah It does say so on the website but the api is still working.

**Comment by somnathrakshit** [comment]
You can use the [Hackerearth API](https://www.hackerearth.com/docs/wiki/developers/v3/) maybe. Overall, a nice project.

**Comment by cB-Abhinav-Gautam** [comment]
Yeah I was thinking of shifting to it when Hackerrank api stops working.
But it still works apparently.

**Comment by somnathrakshit** [comment]
They even have a [Python SDK](https://github.com/HackerEarth/he-sdk-python/blob/master/hackerearth/test_api.py) for it to make things easy.
